css/components/button.css

```diff
@@ -0,0 +1,258 @@
+/* 
+  Buttons
+*/
+
+.btn {
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  gap: var(--space-sm); /* In case of Icon button*/
+  text-decoration: none;
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+  font-size: inherit;
+  text-transform: uppercase;
+
+  /* Design */
+  background-color: var(--primary-color); /*  Default color set to primary */
+  color: inherit;
+  overflow: hidden;
+  outline: none;
+  font-family: inherit;
+  box-shadow: var(--box-shadow-xm);
+  transition: all 250ms var(--ease-in-out);
+
+  /* Sizing and Spacing */
+  padding: var(--space-xs) var(--space-md);
+  font-size: var(--text-md);
+}
+.btn:hover,
+.btn:focus {
+  background-color: var(--primary-light);
+  box-shadow: var(--box-shadow-sm);
+}
+
+.btn-outlined {
+  background-color: transparent !important;
+  border: 3px solid;
+  border-color: var(--primary-color);
+  color: var(--primary-color);
+}
+.btn-outlined:hover,
+.btn-outlined:focus {
+  color: var(--primary-light);
+  border-color: var(--primary-light);
+}
+
+.btn-rounded {
+  border-radius: var(--radius-md);
+}
+
+.btn-action {
+  padding: var(--space-md) !important;
+  aspect-ratio: 1/1;
+  border-radius: 50%;
+}
+
+/* Colors */
+.btn-primary {
+  background-color: var(--primary-color);
+}
+.btn-secondary {
+  background-color: var(--secondary-color);
+}
+.btn-accient {
+  background-color: var(--accient-color);
+}
+.btn-danger {
+  background-color: var(--danger-color);
+}
+.btn-success {
+  background-color: var(--success-color);
+}
+.btn-warn {
+  background-color: var(--warn-color);
+}
+.btn-dark {
+  background-color: var(--dark-color);
+  color: var(--light-color);
+}
+.btn-light {
+  background-color: var(--light-color);
+  color: var(--dark-color);
+}
+.btn-disabled {
+  cursor: none;
```

Comment:
you can use cursor "not-allowed" here.

Reply:
Thanks for the suggestion @rahulmistry751, It definitely makes sense to put "not-allowed" here.